src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -433,11 +433,11 @@ CHIP_ERROR DiscoveryImplPlatform::UpdateCommissionableInstanceName()
     return CHIP_NO_ERROR;
 }
 
-void DiscoveryImplPlatform::HandleDnssdPublish(void * context, const char * type, CHIP_ERROR error)
+void DiscoveryImplPlatform::HandleDnssdPublish(void * context, const char * type, const char * instanceName, CHIP_ERROR error)
 {
     if (CHIP_NO_ERROR == error)
     {
-        ChipLogProgress(Discovery, "mDNS service published: %s", type);
+        ChipLogProgress(Discovery, "mDNS service published: %s; instance name: %s", type, instanceName);
```

Comment:
I see some callbacks setting things as nullptr. Is `%s` safe for nullptr or is that path never setting it?

How about instead of nullptr for instancename, we have some `const char *kUnknownInstanceName="UNKNOWN"` everywhere?  That would server both as code commenting (what does the argument represent) and avoid nullptr errors.

Reply:
I see the nullptr only happens on callback and we also do that on type.
I wish our interface was something like a union of `Error|Data` in some type safe way, so that illegal states can never be represented.

Reply:
nullptr only happens when `error` is an error.  Same thing for the `type`: that's nullptr on errors , and valid otherwise.

This log only happens in the non-error cases, when both strings are valid.

Reply:
Right , reading the code it seems consistent: if you have an error, no other "data" can be valid.

I find it annoying to pass nullptr though and unsightly. Maintainability though does not need fixing for 1.0 (probably even should not ... we are stabilizing).